m4/ax_lua.m4

```diff
@@ -181,7 +181,7 @@
 #   modified version of the Autoconf Macro, you may extend this special
 #   exception to the GPL to apply to your modified version as well.
 
-#serial 42
+#serial 43
```

Comment:
The serial number is the number of commits that modify this macro. This means that every commit must bump it. Otherwise the test suite fails.

Reply:
Fair enough. I wasn't sure how PR history was going to be handled, but this is two distinct operations, one bug fix and one feature, so I did two commits and now have made it two serials.

I'll have a much more extensive addition with LuaJIT support in another PR soon.

Reply:
The test suite doesn't pass for me locally, but all the errors are about *other* macros untouched by this PR. It's hard for me to tell what's going on remotely since the test suite doesn't run in CI when I push (presumably first-time contributor PRs are blocked from automatic CI jobs).